packages/lib-utils/src/app/redux/reducers/k8s/index.ts

```diff
@@ -1,15 +1,271 @@
+  //       https://bugzilla.redhat.com/show_bug.cgi?id=1814822
+  if (obj.apiVersion === 'packages.operators.coreos.com/v1' && obj.kind === 'PackageManifest') {
+    return `(${obj.status?.catalogSource})-${name}`;
+  }
+  return (namespace ? `(${namespace})-` : '') + name;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const moreRecent = (a: any, b: any) => {
+  const metaA = a.get('metadata').toJSON();
+  const metaB = b.get('metadata').toJSON();
+  if (metaA.uid !== metaB.uid) {
+    return new Date(metaA.creationTimestamp) > new Date(metaB.creationTimestamp);
+  }
+  return parseInt(metaA.resourceVersion, 10) > parseInt(metaB.resourceVersion, 10);
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const removeFromList = (list: ImmutableMap<string, any>, resource: K8sResourceCommon) => {
+  const qualifiedName = getQN(resource);
+  consoleLogger.info(`deleting ${qualifiedName}`);
+  return list.delete(qualifiedName);
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const updateList = (list: ImmutableMap<string, any>, nextJS: K8sResourceCommon) => {
+  const qualifiedName = getQN(nextJS);
+  const current = list.get(qualifiedName);
+  const next = fromJS(nextJS);
+
+  if (!current) {
+    return list.set(qualifiedName, next);
+  }
+
+  if (!moreRecent(next, current)) {
+    return list;
+  }
+
+  // TODO: (kans) only store the data for things we display ...
+  //  and then only do this comparison for the same stuff!
+  if (
+    current
+      .deleteIn(['metadata', 'resourceVersion'])
+      .equals(next.deleteIn(['metadata', 'resourceVersion']))
+  ) {
+    // If the only thing that differs is resource version, don't fire an update.
+    return list;
+  }
+
+  return list.set(qualifiedName, next);
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const loadList = (oldList: any, resources: any) => {
+  const existingKeys = new Set(oldList.keys());
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return oldList.withMutations((list: ImmutableMap<string, any>) => {
+    (resources || []).forEach((r: K8sResourceCommon) => {
+      const qualifiedName = getQN(r);
+      existingKeys.delete(qualifiedName);
+      const next = fromJS(r);
+      const current = list.get(qualifiedName);
+      if (!current || moreRecent(next, current)) {
+        list.set(qualifiedName, next);
+      }
+    });
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    existingKeys.forEach((k: any) => {
+      const r = list.get(k);
+      const metadata = r.get('metadata').toJSON();
+      if (!metadata.deletionTimestamp) {
+        consoleLogger.warn(
+          `${metadata.namespace}-${metadata.name} is gone with no deletion timestamp!`,
+        );
+      }
+      list.delete(k);
+    });
+  });
+};
+
+const sdkK8sReducers = (state: K8sState, action: K8sAction): K8sState => {
```

Comment:
This reducer is a mess, and I opt for us not to try and fix that now.

I assume the logic is fairly close to that of OpenShift Console one we migrated?